src/index.ts

```diff
@@ -397,10 +396,9 @@ const createWindow = () => {
   // @ts-expect-error Property 'isMaximized' does not exist on type 'App'.
   app.isMaximized = mainWindow.isMaximized();
 
-  mainWindow.webContents.on('new-window', (e, url) => {
-    e.preventDefault();
-    openExternalUrl(url);
-  });
+  mainWindow.webContents.setWindowOpenHandler(() => ({
```

Comment:
I think the flow here looks wrong.

We should trigger openExternalUrl(url) and send {action: "deny"} right ?

Reply:
yes - could you please make the change?
Also, i dont see the unread message count in the sidebar with this branch.

Reply:
Sure will check

Reply:
yes - also context menu is missing and clicking on links doesn't open the default external browser

Reply:
@balajiv113 - now that we have backported some of the fixes (for deprecation), I see that only the `nodeIntegration` has been removed after the major version upgrade of electron in this PR. Based on this, can we conclude that we need to file a bug report against electron to get this sorted before proceeding with any more upgrades of electron?

Reply:
Yes yes exactly. I created one issue already in electron with a simple reproduction repo as well

https://github.com/electron/electron/issues/36608

Reply:
This has been just released last night in electron v22.0.1. Closing